src/shapes/object.class.js

```diff
@@ -1395,9 +1395,9 @@
       }
       ctx.shadowColor = shadow.color;
       ctx.shadowBlur = shadow.blur * fabric.browserShadowBlurConstant *
-        (multX + multY) * (scaling.scaleX + scaling.scaleY) / 4;
-      ctx.shadowOffsetX = shadow.offsetX * multX * scaling.scaleX;
-      ctx.shadowOffsetY = shadow.offsetY * multY * scaling.scaleY;
+        (multX + multY) * (shadow.shadowUniform ? 1 : scaling.scaleX + scaling.scaleY) / 4;
```

Comment:
scalingX + scalingY when object is not scaled, has a value of 2. i think in this case this number here should be 2.

Reply:
right, good catch!